dotCMS/src/main/java/com/dotcms/rendering/velocity/util/VelocityUtil.java

```diff
@@ -279,203 +277,45 @@ public static ToolboxManager getToolboxManager () {
-
-	        // gets pageChannel for this path
-	        String pageChannel = UtilMethods.getPageChannel(idURI);
-	        context.put("pageChannel", pageChannel);
-	        context.put("PREVIEW_MODE", new Boolean(mode == PageMode.PREVIEW_MODE));
-	        context.put("EDIT_MODE", new Boolean(mode == PageMode.EDIT_MODE));
-	        context.put("ADMIN_MODE", new Boolean(mode.isAdmin));
-
-	        // for publish button on admin control
-	        // I HAVE TO FIX THIS!!!!! FOR THE NEW ONE
-	        context.put("TEMPLATE_LIVE_CONTENT", new Boolean(false));
-	        context.put("CONTAINER_LIVE_CONTENT", new Boolean(false));
-	        context.put("CONTENTLET_LIVE_CONTENT", new Boolean(false));
-
-
-	
-
-	        context.put("livePage", mode.showLive);
-
-	        context.put("language", (String) request.getSession().getAttribute(com.dotmarketing.util.WebKeys.HTMLPAGE_LANGUAGE));
-
-	        if (mode.isAdmin) {
-
-	            // Making sure you are viewing the latest list of languages on any
-	            // admin mode
-	            languages = APILocator.getLanguageAPI().getLanguages();
-	            context.put("languages", languages);
-
-	            // gets user id from request for mod user
-	            com.liferay.portal.model.User backendUser = null;
-
-	            try {
-	                backendUser = com.liferay.portal.util.PortalUtil.getUser(request);
-	                // Skin skin = backendUser.getSkin();
-	                // context.put("USER_SKIN", skin.getSkinId());
-	                context.put("backendUser", backendUser);
-	            } catch (Exception nsue) {
-	                Logger.warn(VelocityUtil.class, "Exception trying yo getUser: " + nsue.getMessage(), nsue);
-	            }
-
-	            HttpSession session = request.getSession();
-	            context.put("directorURL", session.getAttribute(com.dotmarketing.util.WebKeys.DIRECTOR_URL));
-	            context.put("viewFoldersURL", session.getAttribute(com.dotmarketing.util.WebKeys.VIEW_FOLDERS_URL));
-	            context.put("previewPageURL", session.getAttribute(com.dotmarketing.util.WebKeys.PREVIEW_PAGE_URL));
-	            context.put("viewContentsURL", session.getAttribute(com.dotmarketing.util.WebKeys.VIEW_CONTENTS_URL));
-	            context.put("viewBrowserURL", session.getAttribute(com.dotmarketing.util.WebKeys.VIEW_BROWSER_URL));
-
-
-
-	        }
-
-	        context.put("host", host);
-
-	   }
 	
 	   
 
+    /**
+     * This method tries to build a cache key based on information given in the request. Post
+     * requests are ignored and will not be cached.
+     *
+     * @param request - The {@link HttpServletRequest} object.
+     * @return The page cache key if the page can be cached. If it can't be cached or caching is not
+     *         available, returns <code>null</code>.
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    public static String getPageCacheKey(final HttpServletRequest request, final IHTMLPage page)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 5, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1142)